### ISSUE TITLE
Nested prefixes should be appended instead of being overwritten

### ISSUE THREAD

**Issue by rarora7777** [JoshClose/CsvHelper#1968]
**Is your feature request related to a problem? Please describe.**
I want to use nested prefixes. Consider this design where I have a class `Transform` containing a member of type `Vector2` and I provide a `Prefix` for it. Objects of type `Transform` are then used as members of the class `TransformPair` where I provide a `Prefix` again, but this seems to be overwritten with the `Prefix` provided in the `ClassMap` for `Transform`. Is there a way to avoid this? Here's the sample code (or you can run it at https://dotnetfiddle.net/UWd57A)

```csharp
using System;
using System.Globalization;
using System.Collections.Generic;
using System.Text;
using System.IO;
using CsvHelper;
using CsvHelper.Configuration;

public class Vector2
{
	public float x {get; set;}
	public float y {get; set;}
}

public sealed class Vector2Map : ClassMap<Vector2>
{
	public Vector2Map()
	{
		AutoMap(CultureInfo.InvariantCulture);
	}
}

public class Transform
{
	public bool isValid;
	public Vector2 position;
}

public sealed class TransformMap : ClassMap<Transform>
{
	public TransformMap()
	{
		Map(m => m.isValid).Name("IsValid");
		References<Vector2Map>(m => m.position).Prefix("Pos.");
	}
}

public class TransformPair
{
	public Transform left;
	public Transform right;
}

public sealed class TransformPairMap : ClassMap<TransformPair>
{
	public TransformPairMap()
	{
		References<TransformMap>(m => m.left).Prefix("Left.");
		References<TransformMap>(m => m.right).Prefix("Right.");
	}
}


					
public class Program
{
	public static void Main()
	{
		List<TransformPair> list = new List<TransformPair>();

		
		StringBuilder sb = new StringBuilder();
		StringWriter sw = new StringWriter(sb);
		using (var csv = new CsvWriter(sw, CultureInfo.InvariantCulture))
		{
			csv.Context.RegisterClassMap<TransformPairMap>();
			csv.WriteHeader(typeof(TransformPair));

			csv.Flush();
		}
			   
		Console.WriteLine(sb);
	}
}
```

When I run this code, the output is:

`Left.IsValid,Pos.x,Pos.y,Right.IsValid,Pos.x,Pos.y`

**Describe the solution you'd like**
I would like the output to instead be

`Left.IsValid,Left.Pos.x,Left.Pos.y,Right.IsValid,Left.Pos.x,Left.Pos.y`

That is, the prefix specified in `TransformMap` should be appended to the one specified in `TransformPairMap`.

**Describe alternatives you've considered**
Not sure how to achieve this. I would appreciate any pointers to currently available alternatives.

**Additional context**

**Comment by mariusz-schimke** [comment]
Yeah, I noticed this behavior yesterday and find it a bit cumbersome to handle. I think too that prefixes should be passed from the parent through all nesting levels.

**Comment by rarora7777** [comment]
@mariusz-schimke Turns out that this got fixed in version 30.0.0 released a few months after I created this issue: https://github.com/JoshClose/CsvHelper/pull/1506

The `Prefix()` function now takes a second boolean parameter, which can be set to `true` to enable the behaviour I desired.

Check out my updated fiddle: https://dotnetfiddle.net/UWd57A#

**Comment by mariusz-schimke** [comment]
@rarora7777, thanks for your response! I did notice this param as I tested it, but thought it didn't work the way I expected, so I dropped the idea of using it. But now that I played with your fiddle, I noticed that it is the parent that decides whether the child inherits a prefix, not the child. My intuition initially told me that the child map chooses to inherit from the parent map, but I was wrong. But after rethinking it several times it seems logical and makes sense.

I think that I'd rather use the word `propagate` than `inherit` for this scenario, unless my perspective on these inheritance dependencies is still wrong...

Thanks a lot for your help! :)
